pkg/datapath/link/link.go

```diff
@@ -70,12 +55,29 @@ func GetIfIndex(ifName string) (uint32, error) {
 	return uint32(iface.Attrs().Index), nil
 }
 
-type ifNameCache struct {
+type LinkCache struct {
 	mu          lock.RWMutex
 	indexToName map[int]string
 }
 
-func (c *ifNameCache) syncCache() error {
+// NewLinkCache begins monitoring local interfaces for changes in order to
+// track local link information.
+func NewLinkCache() *LinkCache {
```

Comment:
I don't know where this is called from but right now, the way `NewLinkCache` is written, this will behave as a singleton.

Reply:
Ah, because the name for the controller is not unique? Yeah correct. Given that we will only call this constructor once from either the cilium-agent or `cilium monitor ...` commands, this is an OK tradeoff. I can make this more clear in the godoc.

Reply:
^^ I ended up just adding a `sync.Once()` for this so we don't have to document. It means that there's still a package-level variable, but that's not such a big deal now that the LinkCache has a constructor and external packages need to create the LinkCache and use the methods against that object to fetch names.